pkg/client/unversioned/request.go

```diff
@@ -48,6 +48,16 @@ import (
 // are therefore not allowed to set manually.
 var specialParams = sets.NewString("timeout")
 
+// backoff is the underlying backoff struct we use to prevent thundering herd drama, esp when this runs in a controller.
+var backoff = &URLBackoff{Backoff: util.NewBackOff(1*time.Second, 120*time.Second)}
```

Comment:
I would go so far as to say back off *per request*. I don't think we want
to keep state in client objects for this until we're very sure it's
necessary.

On Wed, Nov 18, 2015 at 6:01 AM, Timothy St. Clair <notifications@github.com
> wrote:

> In pkg/client/unversioned/request.go
> <https://github.com/kubernetes/kubernetes/pull/16811#discussion_r45201371>
> :
>
> > @@ -48,6 +48,16 @@ import (
> >  // are therefore not allowed to set manually.
> >  var specialParams = sets.NewString("timeout")
> >
> > +// backoff is the underlying backoff struct we use to prevent thundering herd drama, esp when this runs in a controller.
> > +var backoff = &URLBackoff{Backoff: util.NewBackOff(1*time.Second, 120*time.Second)}
>
> For completeness, here are the details,
>
> timothysc [4:18 PM]
> so you want to just backoff a single client then, without global
> visibility?
> @lavalamp <https://github.com/lavalamp>: ^ It's fine by me I guess.
> Isn't technically accurate for several error conditions but they will
> eventually backoff too
>
> lavalamp [4:36 PM]
> @timothysc <https://github.com/timothysc>: Right. I think they'll all
> back off eventually, and if that's not good enough we can fix it in a
> second pass. But I'd want to see some sort of evidence that it's necessary
> first.
>
> timothysc [4:43 PM]
> Sgtm @jayunit100 <https://github.com/jayunit100> FYI..
>
>    - Just remove global visibility and only backoff per client is the
>    consensus.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16811/files#r45201371>.
>